src/base/bittorrent/session.cpp

```diff
@@ -635,7 +635,7 @@ Path Session::downloadPath() const
 
 bool Session::isValidCategoryName(const QString &name)
 {
-    static const QRegularExpression re(uR"(^([^\\\/]|[^\\\/]([^\\\/]|\/(?=[^\/]))*[^\\\/])$)"_qs);
+    const QRegularExpression re(uR"(^([^\\\/]|[^\\\/]([^\\\/]|\/(?=[^\/]))*[^\\\/])$)"_qs);
```

Comment:
You are missing the point, the static variable is kept in memory **forever** even it is used once.
If you really are concerned about regex recompile issue, you should have a dedicated cache to store them. Using `static` looks like abuse to me.

Reply:
> the static variable is kept in memory **forever** even it is used once.

But it is not a problem for those variables that are repeatedly accessed during application runtime, isn't it? So you can't just blindly reject all static data. Each case requires reflection. Maybe it makes sense somewhere. Although in some cases we could replace global static variables with more local ones.

Reply:
You must misunderstood. I don't strictly reject the use of static variables but I think their usage must be minimized and under close watch.

This code here doesn't seem like used frequently enough IMO, or if you think otherwise, I can make it a class variable.

Reply:
> This code here doesn't seem like used frequently enough

You could check it at least by setting a breakpoint and running qBittorrent under the debugger.
It is often called at application startup, as well as when applying/changing Category filter. Maybe there are some logical flaws in the related code, but that's another problem.

Reply:
>It is often called at application startup

Seems this isn't true. It doesn't run at startup.

>as well as when applying/changing Category filter

2 times when adding a new category.
1 time when removing a category.
1 time when switching categories.

This doesn't count as frequently used IMO. 
Another issue is, I wouldn't expect this validation to run except adding action...

Reply:
> Another issue is, I wouldn't expect this validation to run except at adding action...

This is what I mentioned above:
> Maybe there are some logical flaws in the related code, but that's another problem.

Reply:
As for `QRegularExpression`, there is no clarity for me based on the documentation what is the best way to optimize its use. But nevertheless, it does not seem to me at all a bad idea to cache at least those regular expressions that are used several times in a row and the number of repetitions is scalable depending on runtime factors, for example, on the number of torrents.

Reply:
>all a bad idea to cache at least those regular expressions that are used several times in a row and the number of repetitions is scalable depending on runtime factors, for example, on the number of torrents.

Agree.
I rechecked this PR and there are no uncached instances of it.
The ones that are removed is run on user's demand and doesn't run on start up nor scales with any kind of entries.

Reply:
>Seems this isn't true. It doesn't run at startup.

Seems you have no categories...
Anyway the problem is in misused `isValidCategory()`.